### ISSUE TITLE
Back button

### ISSUE THREAD

**Issue by MoHD20** [fabiospampinato/notable#144]
When jumping between linked notes it'd be nice to have a way to get back to the previous note.

**Comment by MoHD20** [comment]
a Shortcut is kinda sufficient I guess, But I think it's better if we have something like Vim keybindings instead.

**Comment by fabiospampinato** [comment]
There's an issue about that #52

**Comment by MatanRubin** [comment]
+1 for a "Back" functionality, either as a button or as a keyboard shortcut.

**Comment by ShaneG** [comment]
I'd like this as well, and would love it if my back/forward buttons on my mouse and keyboard would work. It is the the same as back/forward in a web browser.

**Comment by KewlCat** [comment]
A big change in the UI ?
There might be not enough space between the search field and the "New" button, but in the button bar above the note editor, maybe there could be a place for two more buttons with "<" and ">" or arrows (whatever you prefer, to mimic your favorite browser) ?
How about : [ edit ] [ tags ] [ attachments ] - [ favorite ] [ pin ] - [ back ] [ forward ] - [ delete ]

(and most importantly, keyboard shortcuts and mouse buttons support, because it's always better if you don't have to move the mouse cursor)

**Comment by fabiospampinato** [comment]
> A big change in the UI ?

That was probably an overstatement.

We could implement this by adding a couple of buttons to the toolbar, as you proposed, but I'd rather remove things from the UI than add them, especially if not everyone needs them.

I think implementing the requested feature is going to take a while and it will probably be implemented like this:

- Firstly I'll add a command palette #338 

- Then I'll add some "Back" and "Forward" commands to it

- Then I'll probably add some default shortcuts for this, since it sounds like a useful feature to be able to trigger quickly

- Then there will be at some point customizable shortcuts #477, so people will be able to bind whatever shortcut they want to it

- Then at some point in the future, when plugins #128 are a thing and the app is much more mature, there will probably be a way to reorganize the buttons available in the toolbar, including the ability to add new ones, remove existing ones and reorder them. At that point in time I could add support for  having those buttons in the toolbar if one so wishes.

**Comment by nrgapple** [comment]
VS code keyboard short cut is 

- <kbd><kbd>alt</kbd>+<kbd>left_arrow</kbd></kbd>: back to previous file
- <kbd><kbd>alt</kbd>+<kbd>right_arrow</kbd></kbd>: forward to file

**Comment by wbgt** [comment]
> When jumping between linked notes it'd be nice to have a way to get back to the previous note.

Yes, the notes are for reading after all, the back and forward buttons are very useful when you have got a lot of inter-linked notes.

**Comment by AldieNightStar** [comment]
Hey. What abotut placing BACK | FORWARD keys over here?
![Screenshot_1](https://user-images.githubusercontent.com/25606423/72531799-e8bbaa80-387a-11ea-9e05-debc091c63f2.png)

**Comment by speyermr** [comment]
My input on this: something that matched the UX of Safari would be perfect, I think:

<img width="490" alt="histmock" src="https://user-images.githubusercontent.com/65182855/85838099-f746b000-b790-11ea-9741-f3f311c91758.png">

**Comment by fabiospampinato** [comment]
> Linked notes are quite difficult to use at the moment. Clicking a link to have a peek at another document is a one-way street and knowing that I won't easily be able to go back makes me wary of following links.

There are tabs in v1.9, those should mostly solve this.

**Comment by nsheff** [comment]
I really like the idea of a back button for navigation. This will make it much easier to navigate interlinked notes, similar to how you would navigate a web page. I would like to be able to use the forward/backward buttons on my mouse to go back and forth, the same way you can do in a web browser. I have a habit of clicking those mouse buttons when browsing the web, and find myself clicking the "back" button a lot inside notable to try to get back to where I was.

The tabs in 1.9 are also super useful, but accomplish a different dynamic.
